### ISSUE TITLE
Disable internal worker when internal worker connect

### ISSUE THREAD

**Issue by jo-so** [matrix-org/synapse#4809]
**Description:**

Would it be possible that Synapse sets `send_federation: false` internally, when a *federation_sender* connects to the replication port? The same with `enable_media_repo: false` and *media_repository*. Maybe it's possible to flip *send_federation* back, when the worker disconnects.

Maybe Synapse could also tell a second instance to not start, if another work of this kind is already running?

**Comment by erikjohnston** [comment]
I agree that its a bit annoying having to manually specify `send_federation` etc, however its important that only *one* process is actually sending federation any one time. This includes during process restarts or if the replication connection drops etc, so would have to be very careful to ensure correct behaviour during those periods. Handling those sorts of things correctly is actually surprisingly complicated to do correctly, so I don't think it's something that will happen unfortunately.

Having said that, I do agree that its not particularly easy or intuitive to set things up correctly, and there is probably quite a few things that we can do to make things better

**Comment by ShadowJonathan** [comment]
Maybe this can be fixed by fixing #9075, and adding an option to have a homeserver spin up an (emergency) federation send worker if the send queue gets fuller than a specific limit.
